### ISSUE TITLE
Issues Running

### ISSUE THREAD

**Issue by ljstrnadiii** [gokceneraslan/dietnet#1]
Hey!

Thanks for building this model. I want to give it a try. When running make in the 1000G dir I get this output:

```
dietnet preprocess genotypes -p affy_samples.20141118.panel -k 5
make: dietnet: Command not found
Makefile:31: recipe for target 'genotypes_x.npy' failed
make: *** [genotypes_x.npy] Error 127
```

Also, your readme is not up to date with your build/install process. I am not quite sure how to run after I build.

**Comment by gokceneraslan** [comment]
I'm working on the Keras implementation, I'll update the readme when I'm done. Thanks.

**Comment by ljstrnadiii** [comment]
cool. thanks. 

Do you happen to use Docker to run this? I am working on running this with docker, but I am having a difficult time gathering all the dependencies.  I wish plink2 was on aptitude...

**Comment by ljstrnadiii** [comment]
@gokceneraslan, I am also having some trouble using this with python 2.7.

**Comment by ljstrnadiii** [comment]
Thanks again @gokceneraslan , 

I have a Docker image running and training of your model. It gets all the dependencies and runs on tf 1.2

I had to change `weight` to `weights` [133]. (https://github.com/gokceneraslan/dietnet/blob/master/dietnet/network.py#L133). Lots of warnings about depreciated functions. 

The only thing that I have had a problem with is tensorboard. I am not sure if it makes sense to drop the tb summaries in the same dir as the checkpoints. Do you get any tb activity before the training is over?

Let me know if you want the Dockerfile.

**Comment by ngphuoc** [comment]
"I had to change weight to weights": Thanks @ljstrnadiii, I got the same error!

**Comment by ljstrnadiii** [comment]
@ngphuoc , were you able to get tensorboard up and running? network.py clearly defines the scalar tensors for loss and accuracy, and all summaries are being merged in train.py, but there is no scalar data available to tensorboard when I launch.

**Comment by ljstrnadiii** [comment]
Hey @gokceneraslan ,

have you had any success training the dietnet? The accuracy does not seem to be increasing at all even though the loss seems to be converging. 

By the way here is the dockerfile if anyone is interested.

```
FROM nvidia/cuda:8.0-cudnn5-devel-ubuntu14.04

# Pick up some TF dependencies
RUN apt-get update && apt-get install -y --no-install-recommends \
        build-essential \
        curl \
        vim \
        wget \
        libfreetype6-dev \
        libpng12-dev \
        libzmq3-dev \
        pkg-config \
        python \
        python-numpy \
	python-dev \
        python-tk \
        rsync \
        software-properties-common \
        unzip \
        && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*

RUN curl -O https://bootstrap.pypa.io/get-pip.py && \
    python get-pip.py && \
    rm get-pip.py

RUN pip --no-cache-dir install \
        ipykernel \
        jupyter \
        matplotlib \
        scipy \
        sklearn \
        pandas \
        Pillow \
        plinkio \
	six \
        && \
    python -m ipykernel.kernelspec


ENV PLINK_VERSION 1.90
ENV PLINK_ZIP plink_linux_x86_64.zip
ENV PLINK_DIR /opt/plink2

RUN set -ex \
    && apt-get update && apt-get install -y unzip curl \
    && curl -fSL "http://s3.amazonaws.com/plink2-assets/plink2_linux_i686_20170702.zip" -o $PLINK_ZIP \
    && mkdir $PLINK_DIR \
    && unzip $PLINK_ZIP -d $PLINK_DIR \
    && rm $PLINK_ZIP


# Install TensorFlow GPU version.
RUN pip --no-cache-dir install \
    https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow_gpu-1.2.0-cp27-none-linux_x86_64.whl

# For CUDA profiling, TensorFlow requires CUPTI.
ENV LD_LIBRARY_PATH /usr/local/cuda/extras/CUPTI/lib64:$LD_LIBRARY_PATH

# symlink for plink2
RUN cd /usr/local/bin \
    && ln -s $PLINK_DIR/plink plink2


# TensorBoard
EXPOSE 6006
# IPython
EXPOSE 8888

COPY . /usr/local/dietnet/
#COPY setup.py /usr/local/dietnet/

#RUN ls /usr/local/ && ls /usr/local/dietnet



# set workdir and make dietnet
WORKDIR "/usr/local/dietnet"
RUN python setup.py install

# nav to bin to run dietnet
WORKDIR  "/usr/local/bin" 

# copy dir
COPY launch.sh /usr/local/bin 
RUN chmod +x /usr/local/bin/launch.sh


ENTRYPOINT ["./launch.sh"]
#RUN ["/bin/bash"]
```

For anyone interested, my .dockerignore file contains:
```
1000G
log
```
and my launch.sh file contains:
```
#!/bin/sh
./dietnet train ../dietnet/1000G/genotypes -l /usr/local/dietnet/log --batchsize 120
```
I build the file with `sudo docker build -t tf_diet_np -f Dockerfile.gpu .` Then, I run it with
```
sudo nvidia-docker run -it -v /local/path/to/dietnet:/usr/local/dietnet tf_diet
```
detach and run
```
sudo nvidia-docker run -it -v /local/path/to/dietnet:/usr/local/dietnet tf_diet_env
```
where tf_diet_env is a docker build where the entry point is bash instead of the launch file. Then, I can run tensorboard in that docker container. Or you can just run the tf_diet container in detach mode with -d instead of -it and launch tensorboard from within the same container.

In order to get tensorboard running, I had to add the step to the add writer. I also had an issue with the setup file and just assign tensorflow to the gpu version without the try, except. 


![loss](https://user-images.githubusercontent.com/3171991/27920350-82dcacf4-6231-11e7-98e6-6bd3639e4c7d.png)
![accur](https://user-images.githubusercontent.com/3171991/27920365-85ce22a8-6231-11e7-9836-106979f57553.png)
